rotkehlchen/exchanges/woo.py

```diff
@@ -0,0 +1,537 @@
+                )
+                return []
+            for entry in entries:
+                try:
+                    result = deserialization_method(entry)
+                except (DeserializationError, UnknownAsset, KeyError) as e:
+                    log.error(
+                        f'Error processing a Woo {case_pretty}.',
+                        entry=entry,
+                        error=str(e),
+                    )
+                    self.msg_aggregator.add_error(
+                        f'Failed to deserialize a Woo {case_pretty}. '
+                        f'Check logs for details. Ignoring it.',
+                    )
+                    continue
+                results.append(result)
+            if len(entries) < API_MAX_LIMIT:
+                break
+            try:
+                if case == 'trades':
+                    call_options['fromId'] = entries[-1]['id']
+                elif (meta := response_dict.get('meta')) is not None and meta['current_page'] * meta['records_per_page'] < meta['total']:  # noqa: E501
+                    call_options['page'] += 1
+            except (KeyError, TypeError) as e:
+                msg = f'The entries do not have the key {e}.' if isinstance(e, KeyError) else f'The response keys have improper types {e}.'  # noqa: E501
+                log.error(
+                    f'Error loading all Woo {case_pretty}. {msg}',
+                    entries=entries,
+                )
+                self.msg_aggregator.add_error(
+                    f'Failed to load all Woo {case_pretty}. Check logs for details.',
+                )
+                break
+        return results
+
+    def _deserialize_asset_movement(self, movement: dict[str, Any]) -> AssetMovement:
+        """
+        Deserialize a Woo asset movement returned from the API to AssetMovement
+
+        May raise:
+        - DeserializationError
+        - UnknownAsset
+        - KeyError
+        """
+        asset = asset_from_woo(movement['token'])
+        fee_currency = asset_from_woo(movement['fee_token'] if movement['fee_token'] != '' else 'USDT')  # noqa: E501
+        movement_id = str(movement['id'])
+        timestamp = Timestamp(int(float(movement['created_time'])))
+        amount = deserialize_asset_amount(movement['amount'])
+        fee = deserialize_fee(movement['fee_amount'])
+        transaction_id = movement['tx_id']
+        if movement['token_side'] == 'DEPOSIT':
+            address = movement['source_address']
+            category = AssetMovementCategory.DEPOSIT
+        else:
+            address = movement['target_address']
+            category = AssetMovementCategory.WITHDRAWAL
+        return AssetMovement(
+            location=Location.WOO,
+            category=category,
+            timestamp=timestamp,
+            address=address,
+            transaction_id=transaction_id,
+            asset=asset,
+            amount=amount,
```

Comment:
As before please move inside here **all** variables that you just made temporary just to populate the constructor. And reduce wasted space in the code.
```suggestion
            amount=deserialize_asset_amount(movement['amount']),
```